jujugui/static/gui/src/app/components/deployment-flow/direct-deploy/direct-deploy.js

```diff
@@ -50,64 +58,104 @@ class DeploymentDirectDeploy extends React.Component {
     });
   }
 
-  render() {
-    let diagram = null;
-    let titleAndDescription = null;
+  /**
+    Generate the entity link.
+
+    @returns {Object} The link markup or null.
+  */
+  _generateLink() {
     const ddEntityId = this.props.ddData.id;
+    let url;
+    try {
+      url = window.jujulib.URL.fromLegacyString(ddEntityId);
+    } catch(_) {
+      url = window.jujulib.URL.fromString(ddEntityId);
+    }
+    url = this.props.generatePath({
+      store: url.path()
+    });
+    if (!url) {
```

Comment:
Can you add a comment as to how this is possible? If this is the `generatePath` from state then I don't think it is possible as it'll always return `${this.baseURL}${path.join('/')}` where the path will contain the string value from the `url.path()` on 75